lookout/style/format/features.py

```diff
@@ -225,17 +254,114 @@ def _classify_vnodes(self, nodes: Iterable[VirtualNode]) -> List[VirtualNode]:
                             y=cls))
         return result
 
+    def _inplace_write_vnode_features(self, vnode: VirtualNode, vnode_focused: VirtualNode,
+                                      row: int, vnode_index: int, features: numpy.ndarray) -> None:
+        """
+        Given a feature `numpy.ndarray`, a `VirtualNode`, a row and a feature index, write the
+        features of the vnode at the correct location in the feature matrix.
+
+        :param vnode: `VirtualNode` we want to write the features of.
+        :param vnode_focused: the focused `VirtualNode` we'll compare our characteristics with
+        :param row: on which row to write the features.
+        :param vnode_index: the column at which we'll start writing the features is \
+                            vnode_index * n_features.
+        :param features: `numpy.ndarray` the features matrix.
+        """
+        vnode_focused_pos = (vnode_focused.start.line, vnode_focused.end.line,
+                             vnode_focused.start.col, vnode_focused.end.col)
+        if vnode is vnode_focused:
+            pos = vnode_focused_pos
+        else:
+            vnode_pos = (vnode.start.line, vnode.end.line, vnode.start.col, vnode.end.col)
+            pos = tuple(map(lambda a, b: abs(a - b), vnode_focused_pos, vnode_pos))
+        role_index = -1
+        if vnode.node:
+            role = vnode.node.internal_type
+            if role in self.roles.ROLE_INDEX:
+                role_index = self.roles.ROLE_INDEX[role]
+        col = vnode_index * len(self.feature_names)
+        features[row, col:col + 4] = pos
+        features[row, col + 4] = role_index
+
+    @staticmethod
+    def _assign_parent(vnode: VirtualNode, vnode_index: int, vnodes: Sequence[VirtualNode],
```

Comment:
better to rename this to `_find_parent()` - we do not really assign anything inside.